pkg/cloudprovider/aws/capacity.go

```diff
@@ -96,25 +86,5 @@ func (c *Capacity) Create(ctx context.Context, packings []*cloudprovider.Packing
 }
 
 func (c *Capacity) GetInstanceTypes(ctx context.Context) ([]cloudprovider.InstanceType, error) {
-	return c.instanceTypeProvider.Get(ctx, c.provisioner.Spec.Cluster)
-}
-
-func (c *Capacity) GetZones(ctx context.Context) ([]string, error) {
```

Comment:
Why did you remove these following three functions? Do we not need them anymore?

Reply:
I reoriented the logic such that everything is registered at process start time. Check out SupportedArchitecture and variants. This is all generated by a single GetInstanceTypes on startup and then traversing that list to compute the set of supported zones,arch,etc. These are registered when the cloudprovider is instantiated (see register.go)

Constraints will then be applied against those supported baselines. Not only does this cut down on API calls, it also was necessary to make the knative webhooks work. Knative webhooks do not allow dependency injection and highly discourage making dynamic decisions, which I agree with, as it can result in nondeterministic behavior. 

As a result, these functions are no longer necessary, since we can use the static Supported variables.

Reply:
Great. Thanks.